core/src/test/java/org/apache/iceberg/metrics/TestScanMetricsResultParser.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.concurrent.TimeUnit;
+import org.apache.iceberg.metrics.ScanReport.ScanMetricsResult;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestScanMetricsResultParser {
+
+  @Test
+  public void nullMetrics() {
+    Assertions.assertThatThrownBy(() -> ScanMetricsResultParser.fromJson((JsonNode) null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse scan metrics from null object");
+
+    Assertions.assertThatThrownBy(() -> ScanMetricsResultParser.toJson(null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid scan metrics: null");
+  }
+
+  @Test
+  public void missingFields() {
+    Assertions.assertThatThrownBy(() -> ScanMetricsResultParser.fromJson("{}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse timer from missing field: total-planning-duration");
+
+    Assertions.assertThatThrownBy(
+            () ->
+                ScanMetricsResultParser.fromJson(
+                    "{\"total-planning-duration\":{\"name\":\"totalPlanningDuration\",\"count\":1,\"time-unit\":\"NANOSECONDS\",\"total-duration\":600000000000}}"))
```

Comment:
Can we suppress the name since it is passed as the key?

Also, we normally use Kebab case like the key here. I think we should make name and the key consistent. It's weird to have both cases.

Reply:
I refactored the code so that we don't have to repeat the metric name and also changed metrics names to use kebab-case